staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -498,8 +503,69 @@ func GetFieldManagerFlag(cmd *cobra.Command) string {
 	return GetFlagString(cmd, "field-manager")
 }
 
+type DryRunStrategy int
+
+const (
+	DryRunNone DryRunStrategy = iota
+	DryRunClient
+	DryRunServer
+)
+
+// TODO(jmodes): remove
 func GetDryRunFlag(cmd *cobra.Command) bool {
-	return GetFlagBool(cmd, "dry-run")
+	dryRunStrategy, err := GetDryRunStrategy(cmd)
+	if err != nil {
+		klog.Fatalf("error accessing --dry-run flag for command %s: %v", cmd.Name(), err)
+	}
+	if dryRunStrategy == DryRunServer {
+		klog.Fatalf("--dry-run=server for command %s is not supported yet", cmd.Name())
+	}
+	return dryRunStrategy == DryRunClient
+}
+
+func GetDryRunStrategy(cmd *cobra.Command) (DryRunStrategy, error) {
+	var dryRunFlag = GetFlagString(cmd, "dry-run")
+	if dryRunFlag == "" && !cmd.Flags().Changed("dry-run") {
+		klog.Warning(`--dry-run is deprecated and can be replaced with --dry-run=client.`)
+		return DryRunClient, nil
+	}
+	b, err := strconv.ParseBool(dryRunFlag)
+	// The flag is not a boolean
+	if err != nil {
+		switch dryRunFlag {
+		case "client":
+			return DryRunClient, nil
+		case "server":
+			return DryRunServer, nil
+		case "none":
+			return DryRunNone, nil
+		default:
+			return DryRunNone, fmt.Errorf(`Invalid dry-run value (%v). Must be "none", "server", or "client".`, dryRunFlag)
+		}
+	}
+	// The flag was a boolean, and indicates true, run client-side
+	if b {
+		klog.Warning(`--dry-run=true is deprecated and can be replaced with --dry-run=client.`)
+		return DryRunClient, nil
+	}
+	klog.Warning(`--dry-run=false is deprecated and can be replaced with --dry-run=none.`)
+	return DryRunNone, nil
+}
+
+// PrintFlagsWithDryRunStrategy sets a success message at print time for the dry run strategy
+//
+// TODO(juanvallejo): This can be cleaned up even further by creating
```

Comment:
Nice

Reply:
I had a WIP to add DryRunFlags to cli-runtime (for both PrintFlags and DryRunStrategy) like the other generic flags, but I'm not sure if we want to do that yet. Also I wasn't sure whether to couple the cli-runtime/DryRunFlags with a DryRunVerifier.